rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java

```diff
@@ -89,7 +89,12 @@ private void doScan(String reader) {
                     intent.putExtras(bundle);
                 }
                 setResult(Activity.RESULT_CANCELED, intent);
-                finish();
+
+                if (com.vijay.jsonwizard.utils.Utils.isEmptyJsonArray(dataArray)) {
+                    super.onBackPressed();
```

Comment:
Move https://github.com/onaio/rdt-standard/pull/961/files#diff-dd323e705143722471da28e8ee65c1b15a10c71d31bc603b624d0f5ee1f67ea7R91 
here.

Reply:
you mean remove this `setResult(Activity.RESULT_CANCELED, intent);`

Reply:
Move it inside the `if `https://github.com/onaio/rdt-standard/pull/961/files#diff-dd323e705143722471da28e8ee65c1b15a10c71d31bc603b624d0f5ee1f67ea7R93

Reply:
done